src/corona_hakab_model/histogram.py

```diff
@@ -0,0 +1,117 @@
+from collections import Counter
+import numpy as np
+from typing import List
+
+from generation.connection_types import ConnectionTypes
+
+
+class Histogram:
+    """
+    This class implements a cumulative normalized histogram, such that each time new data is received,
+    a new normalized histogram is calculated, based on all the accumulated data.
+    The histogram is represented as a dictionary with the following properties:
+     - histogram dict:
+         * 'number_of_connections': The x-axis of the histogram, number of connections for each agent
+         * 'probability_density': The y-axis of the histogram, the probability of an agent the have that number of
+                                  connections.
+         * 'average': The average number of connections for all agents.
+         * 'median': The median number of connections for all agents.
+    """
+    def __init__(self, connections_distribution: List = None):
+        self.weighted_connections = {}
+        self.histogram = {}
+        self.min_connection = 0
+        self.max_connection = 0
+        if connections_distribution is not None:
+            self.update(connections_distribution)
+
+    def __repr__(self):
+        return str(self.histogram)
+
+    def update(self, connections_distribution: List):
+        # weighted_connections is a dictionary where the keys are the number of connections for an agent,
+        # and the values are the weight of that number, i.e. the total number of agents with that number if connections
+        self.weighted_connections = dict(Counter(self.weighted_connections)+Counter(connections_distribution))
+        self.min_connection = np.min(list(self.weighted_connections.keys()))
+        self.max_connection = np.max(list(self.weighted_connections.keys()))
+        probability_density, bins_edges = np.histogram(list(self.weighted_connections.keys()),
+                                                       bins=(self.max_connection - self.min_connection + 1),
+                                                       range=(self.min_connection - 0.5, self.max_connection + 0.5),
```

Comment:
Why this substract and addition?

Reply:
in order to get the bins' centers in the right places, e.g. [1,2,3,4]